components/icon/stories/icon.stories.js

```diff
@@ -90,10 +93,90 @@ export default {
 	},
 };
 
-export const Default = (args) => Template({
-	...args,
-	iconName: args.iconName ?? args.uiIconName,
-	setName: args.setName ?? (args.uiIconName ? "ui" : "workflow"),
-});
+export const Default = (args) => {
+	if (isChromatic()){
+		return TestTemplate({ ...args });
+	} else {
+		return Template({
+			...args,
+			iconName: args.iconName ?? args.uiIconName,
+			setName: args.setName ?? (args.uiIconName ? "ui" : "workflow"),
+		});
+	}
+} 
 
 Default.args = {};
+
+/**
+ * Chromatic VRT template that displays multiple icons to cover various options.
+ */
+const TestTemplate = ({
+	staticColor,
+	customStyles = {},
+	...args
+}) => {
+	const workflow_row_args = [
+		{
+			setName: "workflow",
+			iconName: "Alert",
+			fill: "var(--spectrum-negative-content-color-default)",
+		},
+		{
+			setName: "workflow",
+			iconName: "ArrowLeft",
+		},
+		{
+			setName: "workflow",
+			iconName: "ArrowRight",
+		},
+		{
+			setName: "workflow",
+			iconName: "ArrowUp",
+		},
+		{
+			setName: "workflow",
+			iconName: "ArrowDown",
+		},
+		{
+			setName: "workflow",
+			iconName: "ChevronDown",
+		}
+	];
+
+	return html`
+		${workflow_row_args.map((row_args) => html`
+			<div
+				style=${styleMap({
+					display: "flex",
+					gap: "16px",
+					marginBottom: "16px",
+				})}
+			>
+				${Template({ ...args, ...row_args, size: "xs" })}
+				${Template({ ...args, ...row_args, size: "s" })}
+				${Template({ ...args, ...row_args, size: "m" })}
+				${Template({ ...args, ...row_args, size: "l" })}
+				${Template({ ...args, ...row_args, size: "xl" })}
+			</div>`
+		)}
+		<div
+			style=${styleMap({
+				display: "flex",
+				gap: "16px",
+			})}
+		>
+			${Template({ ...args, setName: "ui", iconName: "Asterisk75", })}
+			${Template({ ...args, setName: "ui", iconName: "Asterisk100", })}
+			${Template({ ...args, setName: "ui", iconName: "Asterisk200", })}
+			${Template({ ...args, setName: "ui", iconName: "Asterisk300", })}
+			${Template({ ...args, setName: "ui", iconName: "Checkmark100", })}
+			${Template({ ...args, setName: "ui", iconName: "ArrowLeft75", })}
+			${Template({ ...args, setName: "ui", iconName: "ArrowLeft100", })}
+			${Template({ ...args, setName: "ui", iconName: "ArrowLeft600", })}
+			${Template({ ...args, setName: "ui", iconName: "ArrowRight200", })}
+			${Template({ ...args, setName: "ui", iconName: "ChevronDown50", })}
+			${Template({ ...args, setName: "ui", iconName: "ChevronDown75", })}
+			${Template({ ...args, setName: "ui", iconName: "ChevronDown500", })}
```

Comment:
The number values at the end of the ui icons are actually stripped and determined dynamically by the sizing in the logic of the template. Maybe a similar sizing to the above?

```suggestion
			${uiIcons.map(iconName => ["xs", "s", "m", "l", "xl"].map(size => Template({ ...args, size, setName: "ui", iconName }))).flat()}
```

Reply:
If the number is provided, it is not stripped out anymore (in utilities, "If the icon name includes its scale, we want to leave that scale"). The UI icons don't use t-shirt sizing and each one supports a different set of number sizing, with some numbers not corresponding to our guesstimate number that's currently appended if no number size was provided; there are more number sizes than t-shirt sizes.

I'm going to make an attempt to address the "UI Icon sizing in Storybook" question in the PR description with conditional controls, to try and clarify this.

I'll also look at simplifying this bit of code with a `map` in a similar way.

Reply:
I've updated this template code in the latest changes. The Chromatic kitchen sink also now renders all the UI icons in their available sizes.